storage_file/models/storage_backend.py

```diff
@@ -79,3 +80,42 @@ def _get_backend_id_from_param(cls, env, param_name, default_fallback=True):
             else:
                 _logger.warn("No backend found, no default fallback found.")
         return backend_id
+
+    @api.depends(
+        "served_by",
+        "base_url",
+        "directory_path",
+        "url_include_directory_path",
+    )
+    def _compute_base_url_for_files(self):
+        for record in self:
+            record.base_url_for_files = record._get_base_url_for_files()
+
+    def _get_base_url_for_files(self):
+        """Retrieve base URL for files."""
+        backend = self.sudo()
```

Comment:
```suggestion
```

Is this necessary?
In this method compute is stored so it already runs on sudo, for the other cases maybe consider `compute_sudo=True`